canonical/canonical.com
# Fix forms generator bugs

Note: form template usage [document](https://webteam.canonical.com/practices/automated-form-builder) should be updated once this is merged

## Done

- Add exception checks when rendering form template
- Use form template on `/data/postgresql` modal
- Added arialabel and placeholder for textarea
- Added exception to remove contact us formfield with `id=about-you` to be excluded from `Comments_from_lead__c`
- Remove unused `modal.js` file
- Apply form modal design as approved by Visual team (similar to https://ubuntu.com/aws#get-in-touch)
- Add inputType fields to `forms-data.json` to accept params: `radio`, `checkbox`, `checkbox-visibility`

## QA

- Go to https://canonical-com-1443.demos.haus/data/postgresql
- Check that secondary navigation is showing (reported bug #1399)
- Click on "Get in touch"
- Modal should pop up
- Check that the formfields are similar to prod https://canonical.com/data/postgresql#get-in-touch
- Check that "Tell us about your use case" textarea has aria label and placeholder text
- Fill up form, submit and see that `Comments_from_lead__c` only includes the comments from "Tell us about your use case" text area


## Issue / Card

Fixes [WD-17343](https://warthogs.atlassian.net/browse/WD-17343) and #1399 

## Screenshots

[if relevant, include a screenshot]


[WD-17343]: https://warthogs.atlassian.net/browse/WD-17343?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
